heft-plugins/heft-lint-plugin/src/SarifFormatter.ts

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import url from 'node:url';
+
+import type * as TEslint from 'eslint';
+
+export interface ISerifFormatterOptions {
+  ignoreSuppressed: boolean;
+  eslintVersion?: string;
+}
+
+export interface ISarifRun {
+  tool: {
+    driver: {
+      name: string;
+      informationUri: string;
+      version?: string;
+      rules: IStaticAnalysisRules[];
+    };
+  };
+  artifacts?: ISarifFile[];
+  results?: ISarifRepresentation[];
+  invocations?: {
+    toolConfigurationNotifications: ISarifRepresentation[];
+    executionSuccessful: boolean;
+  }[];
+}
+
+export interface ISarifRepresentation {
+  level: string;
+  message: {
+    text: string;
+  };
+  locations: ISarifLocation[];
+  ruleId?: string;
+  descriptor?: {
+    id: string;
+  };
+  suppressions?: ISuppressedAnalysis[];
+}
+
+// Interface for the SARIF log structure
+export interface ISarifLog {
+  version: string;
+  $schema: string;
+  runs: ISarifRun[];
+}
+
+interface IRegion {
```

Comment:
```suggestion
export interface IRegion {
```
Referenced from `ISarifPhysicalLocation`